graphql/resolvers.go

```diff
@@ -2846,6 +2846,24 @@ func (r *queryResolver) MainlineCommits(ctx context.Context, options MainlineCom
 	})
 	var mainlineCommits MainlineCommits
 	activatedVersionCount := 0
+
+	mostRecentVersion, err := model.GetMostRecentMainlineCommit(projectId)
+
+	if err != nil {
+		// This shouldn't realy happen, but if it does, we should return an error and log it
+		grip.Warning(message.WrapError(err, message.Fields{
+			"message":   "Error getting most recent version",
+			"projectId": projectId,
+		}))
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error getting most recent mainline commit: %s", err.Error()))
+	}
+
+	// If a commit is the most recent version, we return nil for PrevPageOrderNumber since there aren't any other newer versions
+	if mostRecentVersion.RevisionOrderNumber != versions[0].RevisionOrderNumber {
```

Comment:
I imagine it's unlikely for versions to be empty but for defensive programming we should probably check that versions isn't empty somewhere (before most recent versions maybe)

Reply:
some of these comments haven't been addressed